weave

```diff
@@ -27,27 +36,47 @@ usage() {
     exit 1
 }
 
-[ $(id -u) = 0 ] || {
-    echo "weave must be run as 'root'" >&2
-    exit 1
-}
-
 [ $# -gt 0 ] || usage
 
-if [ "$SCRIPT_VERSION" = "(unreleased version)" ] ; then
-    IMAGE_VERSION=latest
-else
-    IMAGE_VERSION=$SCRIPT_VERSION
-fi
-
-IMAGE_VERSION=${VERSION:-$IMAGE_VERSION}
+# These are potentially used for remote execs, and have defaults, so
+# we introduce them here
+DOCKER_BRIDGE=${DOCKER_BRIDGE:-docker0}
+PROCFS=${PROCFS:-/proc}
 
-BASE_IMAGE=zettio/weave
-BASE_DNS_IMAGE=zettio/weavedns
-BASE_TOOLS_IMAGE=zettio/weavetools
+BASE_IMAGE=$DOCKERHUB_USER/weave
+BASE_DNS_IMAGE=$DOCKERHUB_USER/weavedns
+BASE_EXEC_IMAGE=$DOCKERHUB_USER/weaveexec
 IMAGE=$BASE_IMAGE:$IMAGE_VERSION
 DNS_IMAGE=$BASE_DNS_IMAGE:$IMAGE_VERSION
-TOOLS_IMAGE=$BASE_TOOLS_IMAGE:$IMAGE_VERSION
+EXEC_IMAGE=$BASE_EXEC_IMAGE:$IMAGE_VERSION
+
+exec_remote() {
+    docker run -t --rm \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        -v "$PROCFS":/hostproc \
+        -e VERSION \
+        -e WEAVE_DOCKER_ARGS \
+        -e WEAVE_PASSWORD \
+        -e DOCKER_BRIDGE \
+        -e PROCFS=/hostproc --privileged --net=host \
+        $EXEC_IMAGE --local "$@"
+}
+
+if [ "$1" = "--local" ]; then
+    LOCAL="$1"
+    shift 1
+# Run is a special case because we want to use the 'local' docker to
+# run the container, rather than the docker in $EXEC_IMAGE.
+elif [ "$1" != "run" ]; then
+    exec_remote "$@"
+    exit $?
+fi
+
+[ $(id -u) = 0 ] || [ "$1" = "run" ] || {
```

Comment:
Actually this might be wrong: the idea was that if you're using `run`, the things you need root for are done by weaveexec; but, if you've said `--local`, it'll try to do those directly.